File: system-test/typescript/src/testkit/dsl/port/then/steps/then-coupon.ts
import type { ThenStep } from './then-step.js';

export interface ThenCoupon extends ThenStep<ThenCoupon> {
  hasDiscountRate(discountRate: number): ThenCoupon;
  isValidFrom(validFrom: string): ThenCoupon;
  isValidTo(validTo: string): ThenCoupon;
  hasUsageLimit(usageLimit: number): ThenCoupon;
  hasUsedCount(expectedUsedCount: number): ThenCoupon;
}

File: system/multitier/backend-clean-java/src/testSupport/java/com/mycompany/myshop/backend/testkit/dsl/port/then/steps/ThenCoupon.java
package com.mycompany.myshop.backend.testkit.dsl.port.then.steps;

import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.base.ThenStep;

public interface ThenCoupon extends ThenStep<ThenCoupon> {
    ThenCoupon hasDiscountRate(String expectedDiscountRate);

    ThenCoupon hasDiscountRate(double expectedDiscountRate);

    ThenCoupon hasUsageLimit(int expectedUsageLimit);

    ThenCoupon hasUsedCount(int expectedUsedCount);
}
